### ISSUE TITLE
when using --cb_type dm (cb_explore_adf) warning of resetting to --cb_type ips is incorrect

### ISSUE THREAD

**Issue by marco-rossi29** [VowpalWabbit/vowpal_wabbit#1778]
#### To Reproduce
Data:
[vw_bug.txt](https://github.com/VowpalWabbit/vowpal_wabbit/files/2921313/vw_bug.txt)

#### Expected behavior
As VW is warning that:
warning: cb_type must be in {'ips','dr','mtr'}; resetting to ips.

I expect --cb_type ips to behave exactly the same as --cb_type dm

#### Observed Behavior
```
>vw --cb_explore_adf --epsilon 0.05 -l 0.5 --dsjson vw_bug.txt -P 1 --cb_type ips
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = vw_bug.txt
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
-0.135135 -0.135135            1            1.0    known        0:0.111111...      205
-0.070946 -0.006757            2            2.0    known        0:0.955556...      205
-0.130631 -0.250000            3            3.0    known        2:0.955556...      205
-10.847973 -43.000000            4            4.0    known        4:0.955556...      205

finished run
number of examples = 4
weighted example sum = 4.000000
weighted label sum = 0.000000
average loss = -10.847973               <----------------------------------------
total feature number = 820

>vw --cb_explore_adf --epsilon 0.05 -l 0.5 --dsjson vw_bug.txt -P 1 --cb_type dm
warning: cb_type must be in {'ips','dr','mtr'}; resetting to ips.
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = vw_bug.txt
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
-0.135135 -0.135135            1            1.0    known        0:0.111111...      205
-0.070946 -0.006757            2            2.0    known        0:0.955556...      205
-0.130631 -0.250000            3            3.0    known        2:0.955556...      205
-0.160473 -0.250000            4            4.0    known        0:0.955556...      205

finished run
number of examples = 4
weighted example sum = 4.000000
weighted label sum = 0.000000
average loss = -0.160473               <----------------------------------------
total feature number = 820
```
#### Environment
What version of VW did you use? 8.6.1
What OS or language did you use? Windows command line

**Comment by jackgerrits** [comment]
Thanks for finding this @marco-rossi29! I've submitted a fix in #1779

**Comment by marco-rossi29** [comment]
thanks for fixing!
